gatsby/content/docs/e2ee_cross_signing.mdx

```diff
@@ -0,0 +1,920 @@
+of the keys, until you hit a key that is already verified! This means that given a signature chain as
+follows (assuming you are Alice):
+
+Alice' Master Key --> Alice' User-Signing Key --> Bob's Master Key --> Bob's Self-Signing Key --> Bob's Tablet
+
+And you have previously (via e.g. emoji verification) verified Alice's (your own) master key, and want
+to know if Bob's Tablet is verified, you propagate that chain backwards, provided all the signatures
+are valid, until you hit your verified master Key.
+
+### Verifying a signature
+Libolm to the rescue! It has a method which makes verifying signatures easy! If you have key A and want
+to verify the signature that key B added, you will have to do the following:
+
+1. In the key A `signatures` dict, find the entry for key B and note down the signature.
+2. Create the string for verification: take key A's complete dictionary, strip off of the `unsigned`
+   and `signature` dictionaries and convert it into canonical json.
+3. Use `olmutil.ed25519_verify(key_b_pub_key, canonical_json, signature);` to verify the signature:
+   note that this will throw an error if invalid, and none if valid.
+4. That's it, you have verified the signatrue!
+
+### Check if a *user* is verified
+To see if a user is verified, simply check if their master key is verified! You can find the master
+key of a user by looking for a cross-signing key which has `master` within their `usage` array.
+
+If a user is verified but all of their devices are not, it is a good idea to display a warning.
+
+### Things to look out for
+1. *Never* trust invalid signatures. Just treat them as if there was no signature at all.
+2. Signature loops *will* exist. Add infinite recursion protection to your signature loop checking.
+3. While signature checking doesn't seem to be expensive, it might be a good idea to cache the results
+   in-memory while the program is running.
+4. Be sure to query the keys for a user if you are told in the sync loop they need updating (if you
+   implemented this correctly previously, nothing changes here).
+5. Instead of recursing until you have found a valid key, you could also recurse until you find your own,
+   verified master key. This in practice, however, does not allow easy migration between device-device verification and
+   cross-signing verification. Recursing until you hit a verified device is thus preferred.
+
+## 3. Implementing SSSS
+SSSS is short for Secure Secret Storage and Sharing, which provides a method of storing secret data
+on the server, without the server administrator or anyone else having access to that data. As such,
+this is optimal for storing the private keys for cross-signing on the server so that all your devices
+can use them later on to generate signatures, and thus verify other people.
+
+Unfortunately there doesn't seem to be any help from libolm for implementing SSSS (yet?) so this will
```

Comment:
> (yet?)

Probably not for the foreseeable future, unfortunately.  Adding support for SSSS would mostly end up being just olm providing some cryptographic primitives, and we don't really want olm to be a library of cryptographic primitives (especially as we're looking into changing olm so that you can use better implementations such as HW acceleration).